### PR TITLE
RD-12 radio functions tests

### PR DESCRIPTION
**Pull request by monikahenschke**

### CODE REVIEW

File: src/utils/audio-manager.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+const playMock = jest.fn();
+Audio.prototype.play = async () => playMock();
```
Comment (patryk-bernasiewicz) [inline src/utils/audio-manager.test.js]: Will it work if you just use:

```js
Audio.prototype.play = jest.fn();
```
?

File: src/utils/audio-manager.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+let getAudioManager;
+
+beforeEach(() => {
+  jest.resetModules();
+  getAudioManager = require('./audio-manager').default;
```
Comment (patryk-bernasiewicz) [inline src/utils/audio-manager.test.js]: Don't use `require` in your files. We use ES6 `import`/`export`.

You can import it once at the beginning of your file and it should work just the same (or actually even faster, because you won't import it multiple times).

File: src/utils/audio-manager.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (patryk-bernasiewicz) [inline src/utils/audio-manager.test.js]: React is unnecessary here, you don't use JSX in utils tests

File: src/utils/audio-manager.test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  getAudioManager = require('./audio-manager').default;
+});
+
+test('Check if getAudioManager() has been called twice, is still the same instance of class', () => {
+  const first = getAudioManager();
+  const second = getAudioManager();
+  expect(first === second).toBeTruthy();
+});
+
+test('Check if getAudioManager() has been called with no args then play() returns undefined ', async () => {
```
Comment (patryk-bernasiewicz) [inline src/utils/audio-manager.test.js]: trailing space in test description

File: src/utils/audio-manager.test.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const instanceOfAudioManager = getAudioManager();
+  await instanceOfAudioManager.play();
+
+  expect(playMock).not.toHaveBeenCalled();
+});
+
+test('Check if getAudioManager() has been called with the argument, then play() is called ', async () => {
```
Comment (patryk-bernasiewicz) [inline src/utils/audio-manager.test.js]: trailing space in test description

File: src/utils/audio-manager.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { func } from 'prop-types';
```
Comment (patryk-bernasiewicz) [inline src/utils/audio-manager.js]: Unnecessary import?

File: src/App.js [hunk 7]
```diff
@@ -1,21 +1,31 @@
...
+      <Button type="button" children="Pause" onClick={handlePause} size="big" />
+      <Button type="button" children="Station X " onClick={handleSelect} />
     </div>
   );
-}
-
-function clickPlay() {
-  console.log('play');
-}
+};
```
Comment (patryk-bernasiewicz) [inline src/App.js]: This changes are out of scope of RD-12 ticket

**Review by patryk-bernasiewicz — CHANGES_REQUESTED** [review]

File: src/utils/audio-manager.test.js (hunk 1 above)
Comment (monikahenschke) [reply src/utils/audio-manager.test.js]: it doesn't work, there we have a problem in line 12 audio-manager.js while catching the error:  TypeError: Cannot read properties of undefined (reading 'catch')

File: src/utils/audio-manager.test.js (hunk 2 above)
Comment (monikahenschke) [reply src/utils/audio-manager.test.js]: I've tried to do it this way, but then these tests affect each other. Example:

`test('Check if getAudioManager() has been called with no args then play() returns undefined ', async () => {
  const instanceOfAudioManager = getAudioManager();
  await instanceOfAudioManager.play();

  expect(playMock).not.toHaveBeenCalled();
});

test('Check if getAudioManager() has been called with the argument, then play() is called ', async () => {
  const instanceOfAudioManager = getAudioManager(
    'http://icepool.silvacast.com/COUNTRY108.mp3'
  );
  await instanceOfAudioManager.play();
  expect(playMock).toHaveBeenCalled();
});`

The second test pass, if the first one will skipped, even though I use jest.resetModules() before each test.

File: src/utils/audio-manager.js (hunk 6 above)
Comment (monikahenschke) [reply src/utils/audio-manager.js]: yes :)

**Review by patryk-bernasiewicz — APPROVED** [review]
